gst/nnstreamer/tensor_source/tensor_src_iio.c

```diff
@@ -1612,10 +1612,18 @@ gst_tensor_src_iio_get_caps (GstBaseSrc * src, GstCaps * filter)
   GstTensorSrcIIO *self;
   GstCaps *caps;
   GstPad *pad;
+  GstPadTemplate *pad_template;
 
   self = GST_TENSOR_SRC_IIO (src);
   pad = self->srcpad;
-  caps = gst_pad_get_current_caps (pad);
+  if (GST_IS_PAD (pad)) {
```

Comment:
[suggestion]
caps = gst_pad_get_current_caps (pad);
if (caps == NULL)
    caps = gst_pad_get_pad_template_caps (caps);
}

Reply:
updated, 
also did a bug fix for the pads - 
_get_caps() can be called before _start()
earlier code set srcpad in properties in _start() and used it everywhere which is wrong
updated code to use basesrc->srcpad